pkg/internal/approver/validation/cache.go

```diff
@@ -0,0 +1,53 @@
+package validation
+
+import "sync"
+
+// Cache maintains a cache of compiled validators.
+// The current implementation is a simple lazy cache meaning:
+//
+// 1. Whenever a validator is requested, it first checks the cache.
+// 2. If a compiled validator exists for the supplied CEL expression, it is returned.
+// 3. If the validator doesn't exist in the cache, a new validator is created, compiled, added to the cache, and returned.
+type Cache interface {
+	// Get returns a compiled validator for the supplied CEL expression.
+	// Any compilation errors will be returned to the caller.
+	//
+	// The supplied CEL expression must output a bool.
+	Get(expr string) (Validator, error)
+}
+
+type cache struct {
+	m *sync.Map
```

Comment:
```suggestion
	m sync.Map
```
I don't think this has to be a pointer to a sync.Map.